controllers/capabilities/tenant_internal_reconciler.go

```diff
@@ -309,3 +322,41 @@ func (r *TenantInternalReconciler) retrieveTenantID() (int64, error) {
 
 	return tenantId, nil
 }
+
+func (r *TenantInternalReconciler) isTenantUpdateRequired() bool {
```

Comment:
This method should compare 3scale Tenant with Tenant CR. If any field is not the same, the client should update the 3scale Tenant to match the Tenant CR.

Reply:
Thank you Eguzki. I did more changes now in separate commit. Maybe I'm wrong, but as i see - 
 - portaClient.**ShowTenant** provides Tenant details without optional parameters, and portaClient **UpdateTenant** - **updates  Only Optional parameters**, and not main params.
So on each cicle we're just set optional parameters if they exist in CR. I'm not sure if I could do changes in `3scale-porta-go-client`. But again, maybe I'm wrong, could you please correct me.

Reply:
Yeah, first thing to do is to add new attributes to the Tenant object from the go client library https://github.com/3scale/3scale-porta-go-client/blob/7ec1588d6e5ca4ee87154d8bd2b59feda3a402f2/client/types.go#L429

Reply:
Jira is opened for porta-client - https://issues.redhat.com/browse/THREESCALE-10672

Reply:
Porta-client PR for review: https://github.com/3scale/3scale-porta-go-client/pull/59